### PR TITLE
Item options

### PR DESCRIPTION
**Pull request by ThyWoof**

### CODE REVIEW

**Comment by ThyWoof** [comment]
@Silvergriffon : here is Item Options almost fully migrated. We are still missing the custom Empress Garb textures... Please review these changes to ensure I didn't break anything. I'll add Empress Garb changes later on.

File: SolastaCommunityExpansion/Viewers/Displays/ItemsAndCraftingDisplay.cs [hunk 1]
```diff
@@ -69,92 +71,165 @@ internal static void DisplayItemsAndCrafting()
...
+                        if (toggle)
+                        {
+                            Main.Settings.RecipesInDM.AddRange(ItemCraftingContext.RecipeBooks.Keys);
+                        }
+                    }
+                }
+
+                UI.Label("");
+
+                var keys = ItemCraftingContext.RecipeBooks.Keys;
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Viewers/Displays/ItemsAndCraftingDisplay.cs:215]: You could do this

```
	foreach (var keygroup in keys.Buffer((int)(keys.Count() + 1) / 2))
	{
		using (UI.HorizontalScope())
		{
			foreach (var key in keygroup)
			{
				AddUIForWeaponKey(key);
			}
		}
	}
```

But that would rely on adding System.Interactive Nuget package.

**Review by PhilPJL — APPROVED** [review]

File: SolastaCommunityExpansion/Models/ItemOptionsContext.cs [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        };
+
+        internal static void LoadClothingGorimStock()
+        {
+            if (!Main.Settings.EnableClothingGorimStock)
+            {
+                return;
+            }
+            
+            foreach (ItemDefinition item in DatabaseRepository.GetDatabase<ItemDefinition>().Where(x => x.ArmorDescription.ArmorType == "ClothesType" && !x.Magical && !x.Name.Contains("_Tattoo") && !x.SlotsWhereActive.Contains("TabardSlot")))
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/ItemOptionsContext.cs:112]: Probably a personal style thing but I usually split .Where && clauses like

```
DatabaseRepository.GetDatabase<ItemDefinition>()
	.Where(x => x.ArmorDescription.ArmorType == "ClothesType")
	.Where(x => !x.Magical)
	.Where(x => !x.Name.Contains("_Tattoo"))
	.Where(x => !x.SlotsWhereActive.Contains("TabardSlot")
```

I think that's more readable - although technically slower.

Comment (ThyWoof) [reply SolastaCommunityExpansion/Models/ItemOptionsContext.cs]: agreed it's slower. made 2 lines for readability.

**Comment by ThyWoof** [comment]
migration complete. Empress Garb is in. FYI @Silvergriffon .

File: SolastaCommunityExpansion/Models/ItemOptionsContext.cs [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        {
+            CharacterPresentationBeltOfDwarvenKind.SetOccurencePercentage(Main.Settings.BeltOfDwarvenKindBeardChances);
+            CharacterPresentationBeltOfDwarvenKind.GuiPresentation.SetDescription(Gui.Format("Feature/&AlwaysBeardDescription", Main.Settings.BeltOfDwarvenKindBeardChances.ToString()));
+        }
+
+        internal static void SwitchCrownOfTheMagister()
+        {
+            foreach (var itemPresentation in Crowns.Select(x => x.ItemPresentation))
+            {
+                var maleBodyPartBehaviours = AccessTools.Field(itemPresentation.GetType(), "maleBodyPartBehaviours").GetValue(itemPresentation) as GraphicsCharacterDefinitions.BodyPartBehaviour[];
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/ItemOptionsContext.cs:157]: You can use:
`var maleBodyPartBehaviours = itemPresentation.GetBodyPartBehaviours(CreatureSex.Male)`

File: SolastaCommunityExpansion/Models/ItemOptionsContext.cs [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+                    item.FocusItemDescription.SetFocusType(Main.Settings.EnableMagicStaffFoci ? EquipmentDefinitions.FocusType.Arcane : EquipmentDefinitions.FocusType.None);
+                }
+            }
+        }
+
+        internal static void SwitchRestockAntiquarian()
+        {
+            foreach (var stock in DatabaseHelper.MerchantDefinitions.Store_Merchant_Antiquarians_Halman_Summer.StockUnitDescriptions)
+            {
+                if (!stock.ItemDefinition.Name.Contains("Manual"))
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/ItemOptionsContext.cs:253]: Combine these ifs?

File: SolastaCommunityExpansion/Viewers/Displays/CharacterDisplay.cs [hunk 5]
```diff
@@ -6,7 +6,7 @@ namespace SolastaCommunityExpansion.Viewers.Displays
 {
     internal static class CharacterDisplay
     {
-        private static bool displayFaceUnlockSettings { get; set; } = false;
+        private static bool DisplayFaceUnlockSettings { get; set; } = false;
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Viewers/Displays/CharacterDisplay.cs:9]: Initializing a property to it's default value will generate a warning.

File: SolastaCommunityExpansion/Models/ItemOptionsContext.cs [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+                "LivewoodStaff",
+                "ff3ec29c-734f-4ef6-8d6e-ceb961d9a8a0",
+                "Equipment/&LivewoodStaffTitle",
+                "Equipment/&LivewoodStaffDescription",
+                ComponentPouch_ArcaneAmulet,
+                EquipmentDefinitions.FocusType.Druidic,
+                StaffOfHealing.GuiPresentation.SpriteReference);
+        }
+
+        private static ItemPresentation EmpressGarbOriginalItemPresentation { get; set; } = null;
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/ItemOptionsContext.cs:88]: Initializing a property to it's default value will generate a warning.

**Review by PhilPJL — CHANGES_REQUESTED** [review]
Small changes noted

**Review by PhilPJL — DISMISSED** [review]

**Review by PhilPJL — APPROVED** [review]

**Pull request merged**